### PR TITLE
Fluxify further

### PR DESCRIPTION
**Pull request by jgondron**
-Now that most search objects read data directly from the store, and expect the store to have the data needed by the time they render, there is no reason to propagate these values down. Cleaned up a lot of the props/state values that are now read from the store. In some cases they are still used in a state in order to trigger updates, since it's not recommended to use forceUpdate.
-Cleaned up some unused actions and unused methods in some objects.
-Pulled the row limit into the store and added this to the params when running the query. This will give us a way to change this within the store in the future. I do not have an action for this yet, since it's not needed, but I did test it out using different values and pagination and views seem to work fine with values other than the default of 12.

### CODE REVIEW

File: app/assets/javascripts/components/pages/Search/Search.jsx [hunk 1]
```diff
@@ -84,38 +68,21 @@ var Search = React.createClass({
   },
 
   render: function() {
-    if(!this.state.remoteCollectionLoaded) {
+    // All children of this object expect the collection and all data to be loaded into the SearchStore.
+    // This will prevent mounting them until ready.
+    if(!this.state.readyToRender) {
       return null;
```
Comment (danthewolfe) [inline app/assets/javascripts/components/pages/Search/Search.jsx]: Should we `return (<Loading/>);` here instead of `return null` since this is the top level component?

**Comment by jonhartzler** [comment]
:+1:

**Comment by danthewolfe** [comment]
See comment then :+1:

File: app/assets/javascripts/components/pages/Search/Search.jsx (hunk 1 above)
Comment (jgondron) [inline app/assets/javascripts/components/pages/Search/Search.jsx]: We can, but based on the current state changes, it will only ever show up on the first query and only if that query takes a while to run. If we really wanted a loading component to render, we would also want to trigger this when the SearchStore begins a new query. At that point I would want to render this as an overlay on top of the existing child components, not completely replace it. 

tldr, I can add this for now, it just probably won't have the effect you were thinking.
